src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[INITIAL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
```

Comment:
method `add` should have own realization

Reply:
Is it mentioned somewhere?
According to this article I've intentionally created add() in that way - 

https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html

Don’t create repeating code.
If the logic of your code repeats - move it to the separate private method. For example, the index check for get(), set() and remove() methods should be moved to the separate method.
There is even a software design principle called [DRY](https://dzone.com/articles/software-design-principles-dry-and-kiss) that urges not to repeat yourself.

Reply:
Ok, as you wish. 
Committed. 
:-)